oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StoragePartitioner.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.elasticsearch.base;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.skywalking.apm.util.StringUtil;
+import org.apache.skywalking.oap.server.core.storage.model.Model;
+
+@Slf4j
+public enum StoragePartitioner {
+    /**
+     * The telemetry data, that generated by OAL or MAL, would be partitioned to storage by the functions of the OAL or
+     * MAL. And, the other record data would be insulated storage by themselves definitions.
+     */
+    INSTANCE;
+    private final Base64.Encoder encoder = Base64.getEncoder();
+    public static final String LOGIC_TABLE_NAME = "logic_table";
+
+    /**
+     * The relations of the logic table and the physical table.
+     */
+    private final Map<String, String> tableRelations = new ConcurrentHashMap<>();
```

Comment:
`ConcurrentHashMap`? Is there a race condition?

Reply:
I am not clear about this. If the Create Processor is linear, it should not have a race condition.

Reply:
No race condition, hashmap is enough. This is a small issue. Let's focus on creating process.